src/stats.rs

```diff
@@ -0,0 +1,841 @@
+    /// The user clicked on a link in a message inside Delta Chat
+    InternalLink = 2,
+    /// The user clicked "Paste from Clipboard" in the QR scan activity
+    Clipboard = 3,
+    /// The user clicked "Load QR code as image" in the QR scan activity
+    ImageLoaded = 4,
+    /// The user scanned a QR code
+    Scan = 5,
+}
+
+#[derive(Serialize)]
+struct SecurejoinSources {
+    unknown: u32,
+    external_link: u32,
+    internal_link: u32,
+    clipboard: u32,
+    image_loaded: u32,
+    scan: u32,
+}
+
+#[derive(Debug, Clone, Copy, FromPrimitive, FromSql, PartialEq, Eq, PartialOrd, Ord)]
+enum SecurejoinUIPath {
+    /// The UI path is unknown, or the user didn't open the QR code screen at all.
+    Unknown = 0,
+    /// The user directly clicked on the QR icon in the main screen
+    QrIcon = 1,
+    /// The user first clicked on the `+` button in the main screen,
+    /// and then on "New Contact"
+    NewContact = 2,
+}
+
+#[derive(Serialize)]
+struct SecurejoinUIPaths {
+    other: u32,
+    qr_icon: u32,
+    new_contact: u32,
+}
+
+/// Some information on an invite-joining event
+/// (i.e. a qr scan or a clicked link).
+#[derive(Serialize)]
+struct JoinedInvite {
+    /// Whether the contact already existed before.
+    /// If this is false, then a contact was newly created.
+    already_existed: bool,
+    /// If a contact already existed,
+    /// this tells us whether the contact was verified already.
+    already_verified: bool,
+    /// The type of the invite:
+    /// "contact" for 1:1 invites that setup a verified contact,
+    /// "group" for invites that invite to a group
+    /// and also perform the contact verification 'along the way'.
+    typ: String,
+}
+
+/// Sends a message with statistics about the usage of Delta Chat,
+/// if the last time such a message was sent
+/// was more than a week ago.
+///
+/// On the other end, a bot will receive the message and make it available
+/// to Delta Chat's developers.
+pub async fn maybe_send_stats(context: &Context) -> Result<Option<ChatId>> {
+    if should_send_stats(context).await?
+        && has_time_passed(context, Config::StatsLastSent, SENDING_INTERVAL_SECONDS).await?
+    {
+        let chat_id = send_stats(context).await?;
+
+        return Ok(Some(chat_id));
+    }
+    Ok(None)
+}
+
+pub(crate) async fn maybe_update_message_stats(context: &Context) -> Result<()> {
+    if should_send_stats(context).await?
+        && has_time_passed(
+            context,
+            Config::StatsLastUpdate,
+            MESSAGE_STATS_UPDATE_INTERVAL_SECONDS,
+        )
+        .await?
+    {
+        update_message_stats(context).await?;
+    }
+
+    Ok(())
+}
+
+async fn has_time_passed(context: &Context, config: Config, seconds: i64) -> Result<bool> {
+    let last_time = context.get_config_i64(config).await?;
+    let next_time = last_time.saturating_add(seconds);
+
+    let res = if next_time <= time() {
+        // Already set the config to the current time.
```

Comment:
You wanted to make it one hour in https://github.com/chatmail/core/pull/6851#discussion_r2362929970, but now there are no retries at all. Rare retries won't make harm at least

Reply:
Ok, if we need to make it more random, additional explicit randomization can be added later